### DISCUSSION TITLE
Subscriber was added too many times too quickly, try again later.

### DISCUSSION THREAD

**Discussion by wolispace** [verbb/formie#2172] [Q&A]
I am seeing lots of errors with this message, code 429.

They are all spam, so no need to worry about them.  

Is there (can there be?) a setting to prevent errors like this being logged so the log only contains significant errors (this seems more like a warning than an error)

Thanks

**Comment by engram-design** [comment]
That sounds related to an integration, possibly [Mailchimp](https://github.com/verbb/formie/issues/1990)?

**Comment by wolispace** [comment]
Close, it is Campaign Monitor:

Looking in the error report, I dont readily see which form this came from.

Side question: it would be nice to know which form an error relates to when a site has many forms.

This is an example of the error I am just about to delete, along with 2 others, that arrived this morning.. and Ill be deleting more tomorrow I imagine. 

## Triggering form “campaignMonitor” integration.

API error: “{"Code":429,"Message":"Subscriber was added too many times too quickly, try again later."}” /app/adelaidenationalparkcity/releases/ed2d8bc8bac5c178ee306e1afb0c7bf9248dc667/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:111

```json
{
    "submissionId": 25999,
    "payload": {
        "EmailAddress": "tbqas1rnq6a7wt8@yahoo.com",
        "Name": "xYZORBWQQPNxw",
        "CustomFields": [],
        "Resubscribe": true,
        "RestartSubscriptionBasedAutoresponders": true,
        "ConsentToTrack": "Yes"
    },
    "integration": {
        "apiKey": "$CM_API_KEY",
        "clientId": "$CM_CLIENT_ID",
        "fieldMapping": {
            "Email": "{email}",
            "Name": "{name1}",
            "MobileNumber": "",
            "ConsentToTrack": "Yes",
            "ConsentToSendSms": "",
            "Surname": "",
            "Firstname1": ""
        },
        "listId": "1757903f4661038ae7f13993c57c0aba",
        "name": "Campaign monitor",
        "handle": "campaignMonitor",
        "type": null,
        "sortOrder": 1,
        "cache": [],
        "tokenId": null,
        "uid": "e3cd4bb8-f81d-4502-938a-1e90a7cf7495",
        "optInField": null,
        "referrer": null,
        "ipAddress": "102.36.229.25",
        "id": "1",
        "dateCreated": "2023-04-27T12:27:39+09:30",
        "dateUpdated": "2024-03-13T15:26:05+10:30"
    },
    "description": null
}
```

**Reply by engram-design** [reply]
Yep, it'll record the submission, but not the form that submission was on. I'll see what I can do about adding that to the payload to make things clearer.

I'm not sure if you have any captchas enabled for your form, but I would've thought that prevent "spammy" behaviour before even getting to Formie. But yes, we need to figure out some sort of mechanism to allow or ignore responses. Not _quite_ sure how that's going to look yet.

**Comment by wolispace** [comment]
Great to hear about getting the form into the payload.. will make life much easier for those of us with many forms.

As for how to filter.. seems like user setting of a regex : 'API errors to exclude from the logs' would be a simple/flexible approach.

I would start it off with `/code 249/i` and then add more in as needed.  That could even be a default :-)

**Comment by wolispace** [comment]
Back at work after long xmas break... spammers have been busy:

![image](https://github.com/user-attachments/assets/4a5aba65-5c3c-4cff-bd64-0047fb7c0dfb)

These are all failures due to multiple submissions.

Would be great if we could configure Formie to not report these as Failed as they are known 'junk'.  Will make it easier to see real issues.

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.6](https://github.com/verbb/formie/releases/tag/4.0.0-beta.6).
